port_ocean/core/event_listener/polling.py

```diff
@@ -69,20 +82,17 @@ async def resync() -> None:
             integration = await ocean.app.port_client.get_current_integration()
             last_updated_at = integration["updatedAt"]
 
-            should_resync = (
-                self._last_updated_at is not None
-                or self.event_listener_config.resync_on_start
-            ) and self._last_updated_at != last_updated_at
-
-            if should_resync:
+            if self.should_resync(last_updated_at):
+                await self._before_resync()
                 logger.info("Detected change in integration, resyncing")
-                self._last_updated_at = last_updated_at
+                ocean.app.set_last_updated_at(last_updated_at)
                 running_task: Task[Any] = get_event_loop().create_task(
                     self.events["on_resync"]({})  # type: ignore
                 )
                 signal_handler.register(running_task.cancel)
 
                 await running_task
+                await self._after_resync()
```

Comment:
Yes it will not stay up and running, I think whatever error that will come up to this it will not exit the app.
But we need to take into account that if this request fails in polling event-listener we might run an un called for resync since the latest `updatedAt` will not inline with the current one